sanitizers/template.go

```diff
@@ -0,0 +1,41 @@
+package sanitizers
+
+import (
+	htmlTemplate "html/template"
+	"io"
+	"strings"
+	textTemplate "text/template"
+	"text/template/parse"
+)
+
+const evilTemplateCommand = "{{ .EvilCommand }}"
+
+func HtmlTemplateExecute(hookId int, tmpl *htmlTemplate.Template, wr io.Writer, data any) error {
+	checkForEvilActionAndGuideFuzzer(hookId, tmpl.Tree)
+	return tmpl.Execute(wr, data)
+}
+
+func HtmlTemplateExecuteTemplate(hookId int, tmpl *htmlTemplate.Template, wr io.Writer, name string, data any) error {
+	checkForEvilActionAndGuideFuzzer(hookId, tmpl.Lookup(name).Tree)
+	return tmpl.ExecuteTemplate(wr, name, data)
+}
+
+func TextTemplateExecute(hookId int, tmpl *textTemplate.Template, wr io.Writer, data any) error {
+	checkForEvilActionAndGuideFuzzer(hookId, tmpl.Tree)
+	return tmpl.Execute(wr, data)
+}
+
+func TextTemplateExecuteTemplate(hookId int, tmpl *textTemplate.Template, wr io.Writer, name string, data any) error {
+	checkForEvilActionAndGuideFuzzer(hookId, tmpl.Lookup(name).Tree)
+	return tmpl.ExecuteTemplate(wr, name, data)
+
+}
+
+func checkForEvilActionAndGuideFuzzer(hookId int, tree *parse.Tree) {
+	tmplText := tree.Root.String()
+	if strings.Contains(tmplText, evilTemplateCommand) {
+		ReportFinding("Template Injection")
+	} else {
+		GuideTowardsContainment(tmplText, evilTemplateCommand, hookId)
+	}
+}
```

Comment:
LGTM. The SQL PR should be reworked first to e.g. apply changes that affect all PRs like the typo in `repot.go`.

Only little nitpicky thing I find a little ugly is that we now across als sanitizers have such differing but long function names for essentially the same functionality:

* `checkForEvilCommandAndGuideFuzzer` -> `exec.go`
* `isSyntaxError` + `GuideTowardsContainment` -> `sql.go`
* `checkForEvilActionAndGuideFuzzer` -> `template.go`

Furthermore, sometimes the `ReportFinding` call is within these helper functions like in the case of the template-sanitizer, whereas e.g. for the SQL-sanitizer the call to `ReportFinding` is directly embedded in each hook. Shouldn't we streamline this so future additions have an easier time integrating new sanitizers as they just can follow a sound pattern that we introduced at the beginning?

Reply:
I've refactored the detection and reporting logic so that all detectors now have to implement the `Detector` interface. This should make the overall logic easier to understand and unified among all sanitizers. Could you have a look?

Reply:
That looks a lot cleaner to me! I like the idea of having that common interface. With the exception of the nitpick this LGTM :)